rfcs/rfc0018.md

```diff
@@ -0,0 +1,88 @@
+# RFC - enhanced regex /xx
+
+## Preamble
+
+    Author:  Karl Williamson <khw@cpan.org>
+    ID:      KHW-0001
+    Status:  Draft
+
+## Abstract
+
+Let programmers improve the readability of regular expression patterns beyond
+what is possible now.
+
+## Motivation
+
+Regular expression patterns were designed for concision rather than clarity.
+
+The /x regular expression pattern modifier was created to enable adding
+comments and white space to patterns to make them more readable.  It suffers
+from not working for bracketed character classes, and silently compiling to
+something unintended when the programmer forgets to mark literal white space,
+and much worse, literal '#'.  This last silently swallows the rest of the line
+that was supposed to be a part of the pattern.
+
+I eventually added /xx to at least allow tabs and blanks inside bracketed
+character classes.  This allows a very minor improvement in their readability.
+I could not figure out a way to extend this to allow comments and multiple
+lines inside such a class without making it even more likely that the pattern
+would silently compile to something unintended.  But now, I think this RFC
+fixes that.
+
+## Specification
+
+I propose adding a new opt-in feature.  Call it, for now, "feature
+enhanced_re_xx".  Within its scope, the /xx modifier would change things so
+that inside a bracketed character class [...], any vertical space would be
+treated as a blank, essentially ignored.  Any unescaped '#' would begin a
+comment that ends at the end of the line.
+
+This would change the existing /x behavior where the portion of the line after
+the '#' is parsed, looking for a potential pattern terminating delimiter.
+Under this feature to terminate a pattern, do so before any '#' on a line.
+If an unescaped terminating delimiter is found after a '#' on a line, a warning
+would be raised.
+
+And an unescaped '#' within a comment would raise a warning.  So
```

Comment:
Is this meant to be "unescaped "#" within a *character class*"?  You example has no # within comments, unless `$a[0]` includes one, which you don't mention.

Reply:
within a comment in the pattern is what I mean.  The line

`! @ # . *   # And certain punctuation`

contains  a comment beginning immediately after the `@ `.  The comment is precisely: 

`  . *   # And certain punctuation`

It is easy for a human reader to miss.